taroscript/send.go

```diff
@@ -0,0 +1,164 @@
+package taroscript
+
+import (
+	"errors"
+	"fmt"
+	"math"
+
+	"github.com/lightninglabs/taro/asset"
+	"github.com/lightninglabs/taro/commitment"
+	"github.com/lightninglabs/taro/taropsbt"
+)
+
+var (
+	// ErrInvalidCollectibleSplit is returned when a collectible is split
+	// into more than two outputs.
+	ErrInvalidCollectibleSplit = errors.New(
+		"fund: invalid collectible split",
+	)
+
+	// ErrInvalidChangeOutputLocation is returned when the change output is
+	// not at the expected location (index 0).
+	ErrInvalidChangeOutputLocation = errors.New(
+		"fund: invalid change output location, should be index 0",
+	)
+
+	// ErrInvalidSplitAmounts is returned when the split amounts don't add
+	// up to the amount of the asset being spent.
+	ErrInvalidSplitAmounts = errors.New(
+		"fund: invalid split amounts, sum doesn't match input",
+	)
+)
+
+// PrepareOutputAssets prepares the assets of the given outputs depending on
+// the amounts set on the transaction. If a split is necessary (non-interactive
+// or partial amount send) it computes a split commitment with the given input
+// and spend information. The input MUST be checked as valid beforehand and the
+// change output is expected to be declared as such (and be at index 0).
+func PrepareOutputAssets(input *taropsbt.VInput,
+	outputs []*taropsbt.VOutput) error {
+
+	// This should be caught way earlier but just to make sure that we never
+	// overflow when converting the input amount to int64 we check this
+	// again.
+	inputAsset := input.Asset()
+	if inputAsset.Amount > math.MaxInt64 {
+		return fmt.Errorf("amount int64 overflow")
+	}
+
+	// A collectible cannot be split into individual pieces. So there can
+	// only be a tombstone and a recipient output.
+	if inputAsset.Type == asset.Collectible && len(outputs) > 2 {
```

Comment:
I think we only need to enforce this for the case where `Interactive` is false?

Reply:
Yes, updated.